test/integration/node/lifecycle_test.go

```diff
@@ -297,3 +303,30 @@ func TestTaintBasedEvictions(t *testing.T) {
 		})
 	}
 }
+
+// WaitForPodCondition waits a pods to be matched to the given condition.
+func WaitForPodCondition(c clientset.Interface, ns, podName, desc string, timeout time.Duration, condition podCondition, t *testing.T) error {
+	t.Logf("Waiting up to %v for pod %q in namespace %q to be %q", timeout, podName, ns, desc)
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(poll) {
+		pod, err := c.CoreV1().Pods(ns).Get(context.TODO(), podName, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				t.Logf("Pod %q in namespace %q not found. Error: %v", podName, ns, err)
+				return err
+			}
+			t.Logf("Get pod %q in namespace %q failed, ignoring for %v. Error: %v", podName, ns, poll, err)
+			continue
+		}
+		// log now so that current pod info is reported before calling `condition()`
+		t.Logf("Pod %q: Phase=%q, Reason=%q, readiness=%t. Elapsed: %v",
+			podName, pod.Status.Phase, pod.Status.Reason, podutil.IsPodReady(pod), time.Since(start))
```

Comment:
i think we should think about this   `podutil.IsPodReady` , waiting for @oomichi  confirmation. :smile:

Reply:
I cannot follow @tanjunchen comment here..
Can you clarify your comment?

Reply:
sorry,i misunderstand. `podutil.IsPodReady` can be used here (test/integration)